python/pyarrow/types.pxi

```diff
@@ -2157,7 +2151,7 @@ cdef DataType ensure_type(object ty, c_bool allow_none=False):
         return None
     elif isinstance(ty, DataType):
         return ty
-    elif isinstance(ty, six.string_types):
+    elif isinstance(ty, str):
```

Comment:
here, string_types is replaces with only `str`, but just above (in the diff), it is replaced with `(bytes, str)`. Is there a reason for the difference?

Reply:
It seems to me that passing datatypes as bytestrings shouldn't be accepted. As for the other instances in this file, those are about other types, and I wasn't sure so preferred to be conservative.

Reply:
Sounds good